numbers.go

```diff
@@ -36,7 +36,7 @@ type StringOrNumber string
 
 // UnmarshalJSON implements json.Unmarshaler
 func (n *StringOrNumber) UnmarshalJSON(data []byte) error {
-	if len(data) > 2 && data[0] == '"' && data[len(data)-1] == '"' {
+	if len(data) >= 2 && data[0] == '"' && data[len(data)-1] == '"' {
```

Comment:
thanks, but could you please add a test for it?

Reply:
1. agree the test is needed.
2. What about NumberOrString,  it has same issue with empty string, should I fix it too and provide the test ?